server/plugins/AdvancedWebhook.json

```diff
@@ -0,0 +1,50 @@
+{
+  "name": "AdvancedWebhook",
+  "description": "Advanced HTTP post webhook",
+  "disable": false,
+  "trigger": "message",
+  "scope": "after",
+  "config": [
+    {
+      "name": "URL",
+      "label": "Webhook URL",
+      "description": "Destination webhook URL",
+      "type": "text",
+      "required": true
+    },
+    {
+      "name": "contentType",
+      "label": "Content Type",
+      "description": "Content type header to use (eg - application/json)",
+      "type": "text",
+      "required": true
+    },
+    {
+      "name": "filterMode",
+      "label": "Filter Mode",
+      "description": "Filter mode to use (all messages, all defined aliases, per alias only)",
+      "type": "select",
+      "options": [
+        {"value": "0", "text": "Per alias only"},
+        {"value": "1", "text": "Defined aliases"},
+        {"value": "2", "text": "All messages"}
+      ],
+      "required": true
+    },
+    {
+      "name": "contentTemplate",
+      "label": "Content Template",
+      "description": "Template of content to send. /<field>/ will be replaced with the contents for the message (eg - /message/ will get replaced with the message text)",
```

Comment:
May want to provide a list of available fields so users don't have to reference code if they forget syntax

Reply:
Gonna throw in Model layer there. This way we could pull available fields from there, including plugin-generated ones 😏

Reply:
I can add it to the description but was wary of the length. I can write a documentation page but there isn't really a place to put one yet :-P

Let me know if you want it added to the description